File: app/src/main/java/com/alphacorporations/givememymoney/View/ListDebtActivity.java
package com.alphacorporations.givememymoney.View;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.ViewPager;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.alphacorporations.givememymoney.R;
import com.alphacorporations.givememymoney.ViewModel.Injection;
import com.alphacorporations.givememymoney.ViewModel.ViewModelFactory;
import com.alphacorporations.givememymoney.model.Debt;
import com.alphacorporations.givememymoney.model.MainViewModel;
import com.alphacorporations.givememymoney.model.database.DebtDatabase;
import com.alphacorporations.givememymoney.model.database.dao.DebtDao;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import org.greenrobot.eventbus.EventBus;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;

public class ListDebtActivity extends AppCompatActivity implements DebtAdapter.DeleteTaskListener {

    // UI Components
    @BindView(R.id.add_debt)
    FloatingActionButton mFloatingActionButton;

    DebtDatabase mDatabase;
    DebtDao debtDao;
    MainViewModel mMainViewModel;

    private DebtAdapter adapter;
    private List<Debt> mDebtList = new ArrayList<>();



    @SuppressWarnings("NullableProblems")
    @NonNull
    private RecyclerView listDebts;

    @SuppressWarnings("NullableProblems")
    @NonNull
    private TextView lblNoTasks;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mDatabase = DebtDatabase.getInstance(this);
        debtDao = mDatabase.debtDao();

        this.configureViewModel();
        this.initList();

        setContentView(R.layout.activity_main);

        listDebts = findViewById(R.id.list_money);
        lblNoTasks = findViewById(R.id.lbl_no_task);
        listDebts.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        listDebts.setAdapter(adapter);

        listDebts = findViewById(R.id.list_money);
        lblNoTasks = findViewById(R.id.lbl_no_task);

        mFloatingActionButton = findViewById(R.id.add_debt);
        mFloatingActionButton.setOnClickListener(v -> addDebt());

    }


    private void initList() {
        populateList();
        adapter = new DebtAdapter(mDebtList, this, mDatabase);
    }

    @OnClick(R.id.add_debt)
    void addDebt() {
        ActivityCompat.startActivity(this, new Intent(this, AddDebtActivity.class), null);
    }

    private void populateList() {
        final Observer<List<Debt>> debtObserver = debtList -> {
            if (debtList != null) {
                mDebtList = debtList;
                updateTasks();
            }
        };
        this.mMainViewModel.getDebts().observe(this, debtObserver);
    }

    private void configureViewModel() {
        ViewModelFactory mViewModelFactory = Injection.provideViewModelFactory(this);
        this.mMainViewModel = ViewModelProviders.of(this, mViewModelFactory).get(MainViewModel.class);
    }

    @Override
    public void onDeleteDebt(Debt debt) {
        mDebtList.remove(debt);
        this.mMainViewModel.deleteDebt(debt.getId());
        updateTasks();
    }

    private void updateTasks() {
        if (mDebtList.size() == 0) {
            lblNoTasks.setVisibility(View.VISIBLE);
            listDebts.setVisibility(View.GONE);
        } else {
            lblNoTasks.setVisibility(View.GONE);
            listDebts.setVisibility(View.VISIBLE);

            adapter.updateTasks(mDebtList);
        }
    }
}

File: app/src/main/java/com/alphacorporations/givememymoney/View/ListDebtActivity.kt
package com.alphacorporations.givememymoney.View

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import butterknife.OnClick
import com.alphacorporations.givememymoney.R
import com.alphacorporations.givememymoney.event.DeleteMeetingEvent
import com.alphacorporations.givememymoney.model.Debt
import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import kotlinx.android.synthetic.main.activity_main.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe


class ListDebtActivity : AppCompatActivity() {

    //GLOBAL VARIABLES
    var debtList: MutableList<Debt> = mutableListOf()
    lateinit var adapter: DebtAdapter
    private val db = Firebase.firestore
    private lateinit var collections:CollectionReference


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        collections = db.collection("DebtList")
        initList(collections)
        setAdapter()

        add_debt.setOnClickListener { addDebt() }
    }

     fun initList(collection: CollectionReference? = null) {

        collection!!.get().addOnSuccessListener { result ->
            for (document in result) {
                Log.d(Context.ACTIVITY_SERVICE, "${document.id} => ${document.data}")
                val img = document.data["img"].toString()
                val name = document.data["name"].toString()
                val reason = document.data["reason"].toString()
                val date = document.data["date"].toString()
                val amount = document.data["amount"].toString()

                debtList.add(Debt(document.id, img, name, reason, date, amount.toLong()))
            }
            updateTasks()
        }
                .addOnFailureListener { exception ->
                    Log.w(Context.ACTIVITY_SERVICE, "Error getting documents.", exception)
                }
    }

    private fun setAdapter() {
        val mLayoutManager = LinearLayoutManager(this)
        list_money.layoutManager = mLayoutManager
        list_money.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        adapter = DebtAdapter(debtList)
        list_money.adapter = DebtAdapter(debtList)
    }

    @OnClick(R.id.add_debt)
    fun addDebt() {
        ActivityCompat.startActivity(this, Intent(this, AddDebtActivity::class.java), null)
    }

     fun updateTasks() {
        if (debtList.isEmpty()) {
            lbl_no_task.visibility = View.VISIBLE
            list_money.visibility = View.GONE
        } else {
            lbl_no_task.visibility = View.GONE
            list_money.visibility = View.VISIBLE
        }
    }



    override fun onPostResume() {
        super.onPostResume()
        initList()
    }
}